src/Generators/ModelflakeGenerator.php

```diff
@@ -8,17 +8,41 @@
  * Class ModelflakeGenerator
  *
  * Custom implementation inspired by Twitter Snowflake.
- * Generate Snowflake IDs with a 7-bit prefix identifier for the model that the ID was generated for
+ * Generate Snowflake IDs with a prefix identifier for the model that the ID was generated for
  */
-class ModelflakeGenerator implements ISnowflakeGenerator
-{
-    public function __construct(int $prefix)
-    {
-        // TODO: Implement __construct() method.
-    }
-
-    public function generate(): int
-    {
-        // TODO: Implement generate() method.
-    }
+class ModelflakeGenerator implements ISnowflakeGenerator{
+
+	private string $modelName;
+	private int $microtimeLength;
+	private int $identifierLength;
+
+	public function __construct($modelName){
+
+		$this->modelName = $modelName;
+		$this->microtimeLength = 39;
+		$this->identifierLength = 24;
```

Comment:
If you're not passing values in through the constructor, they should simply be set as default values where the variables are declared